### PR TITLE
feat: add DefinitionList and HelpMark

### PR DESCRIPTION
**Pull request by Raubzeug**

### CODE REVIEW

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/DefinitionList/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export {DefinitionList} from './DefinitionList';
+export type {DefinitionListProps, DefinitionListItem} from './types';
```
Comment (amje) [inline src/components/DefinitionList/index.ts:2]: DefinitionListItem should be a component, not a type. Let's also reexport `DefinitionListItemProps`

File: src/components/DefinitionList/types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import type React from 'react';
+
+import type {HelpMarkProps} from '../HelpMark';
+import type {QAProps} from '../types';
+export type DefinitionListItemNote = string | HelpMarkProps;
+
+export interface DefinitionListItem {
```
Comment (amje) [inline src/components/DefinitionList/types.ts:7]:
```suggestion
export interface DefinitionListItemProps {
```

File: src/components/DefinitionList/utils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import type React from 'react';
+
+import {block} from '../utils/cn';
+
+export const b = block('definition-list');
```
Comment (amje) [inline src/components/DefinitionList/utils.ts:5]: Let's move this into `constants.ts`

File: src/components/HelpMark/HelpMark.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import './HelpMark.scss';
+
+const b = block('help-mark');
+const ICON_SIZE = 16;
+
+export interface HelpMarkProps extends QAProps {
+    buttonProps?: React.ButtonHTMLAttributes<HTMLButtonElement>;
+    buttonRef?: React.RefObject<HTMLButtonElement>;
+    delayClosing?: number;
```
Comment (amje) [inline src/components/HelpMark/HelpMark.tsx:19]: Let's keep it unconfigurable for now

File: src/components/HelpMark/HelpMark.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    ...rest
+}: HelpMarkProps) {
+    return (
+        <Popover
+            {...rest}
+            delayClosing={delayClosing}
+            className={b(null, className)}
+            content={children}
+        >
+            <button
```
Comment (amje) [inline src/components/HelpMark/HelpMark.tsx:40]: Can we use Popover's render function children here? That would not add extra element

File: src/components/HelpMark/README.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+```tsx
+import {HelpMark} from '@gravity-ui/uikit';
+```
+
+Component to display help icon with popover
+
+## Examples
+
+Component with rendered raw html (use `content` for plain text) and close on mouse leave after timeout set by `delayClosing`:
```
Comment (amje) [inline src/components/HelpMark/README.md:15]: This description is outdated now

File: src/components/HelpMark/README.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  <a href="https://example.com" target="_blank" rel="noreferrer">
+    amet
+  </a>
+  , at scelerisque suspendisse
+</HelpMark>
+```
+
+<!--/GITHUB_BLOCK-->
+
+Popover with JSX component as content:
```
Comment (amje) [inline src/components/HelpMark/README.md:73]: This text is also misleading, while API has changed

File: src/components/HelpMark/__stories__/HelpMark.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                        // aria-labelledby id is valid after tooltip content is rendered
+                        selector: 'button[aria-labelledby="helpMarkWithoutActionsId"]',
+                    },
+                ],
+            },
+        },
+    },
+} as Meta<typeof HelpMark>;
+
+const DefaultTemplate: StoryFn<HelpMarkProps> = (args) => <HelpMark {...args} />;
```
Comment (amje) [inline src/components/HelpMark/__stories__/HelpMark.stories.tsx:34]: Let's write the stories in CSF3 format

**Review by amje — CHANGES_REQUESTED** [review]

File: CODEOWNERS [hunk 9]
```diff
@@ -1,61 +1,63 @@
...
-/src/components/Tabs @sofiushko
-/src/components/Text @IsaevAlexandr
-/src/components/TreeList @IsaevAlexandr
-/src/components/TreeSelect @IsaevAlexandr
-/src/components/controls/TextArea @korvin89
-/src/components/controls/TextInput @korvin89
-/src/components/Toaster @ogonkov
-/src/components/Tooltip @amje
-/src/components/layout @IsaevAlexandr
+-   @amje @ValeraS @korvin89
```
[the remark this answers is not in the document]
Comment (Raubzeug) [reply CODEOWNERS:1]: linter did it unexpectedly

File: src/components/DefinitionList/__stories__/DefinitionList.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+        children: <Label>label</Label>,
+    },
+];
+const definitionListItems = items.map(({children, ...rest}, index) => (
+    <DefinitionList.Item key={index} {...rest}>
+        {children}
+    </DefinitionList.Item>
+));
+
+export default {
```
[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/components/DefinitionList/__stories__/DefinitionList.stories.tsx:143]: I've converted stories to CSF3 format. Lets improve stories content in a separate PR.

File: src/components/DefinitionList/DefinitionList.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+function prepareChildren(children: React.ReactNode) {
+    const items = React.Children.toArray(children);
+
+    const normalizedItems = [];
+    for (const item of items) {
+        const isItem = isDefinitionListItem(item);
+        if (isItem) {
+            normalizedItems.push(item);
+        } else {
+            warnOnce('Only <DefinitionList.Item> components is allowed inside <DefinitionList>');
```
Comment (amje) [inline src/components/DefinitionList/DefinitionList.tsx:50]:
```suggestion
            warnOnce('[DefinitionList] Only <DefinitionList.Item> components is allowed as children');
```

File: src/components/DefinitionList/index.ts [hunk 12]
```diff
@@ -0,0 +1,2 @@
+export {DefinitionList} from './DefinitionList';
```
Comment (amje) [inline src/components/DefinitionList/index.ts:1]: Let's reexport `DefinitionListItem` component as well. There might be issues with dot notation in Next.js applications

File: src/components/HelpMark/README.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+  </a>
+  , at scelerisque suspendisse
+</HelpMark>
+```
+
+<!--/GITHUB_BLOCK-->
+
+## Properties
+
+| Property    | Type                                            | Required | Default             | Description                                     |
```
Comment (amje) [inline src/components/HelpMark/README.md:72]: Let's use columns in this order and names: `Name, Description, Type, Default`, no Required column, Type and Default should be center aligned

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**